app/src/main/java/com/futurice/freesound/feature/home/HomeViewModel.java

```diff
@@ -28,30 +28,28 @@
 
 import static com.futurice.freesound.utils.Preconditions.get;
 import static io.reactivex.android.schedulers.AndroidSchedulers.mainThread;
-import static io.reactivex.schedulers.Schedulers.computation;
 
 final class HomeViewModel extends BaseViewModel {
 
     @NonNull
     private final Navigator navigator;
 
     @NonNull
-    private final PublishSubject<Unit> openSearch = PublishSubject.create();
+    private final PublishSubject<Unit> openSearchSubject = PublishSubject.create();
 
     HomeViewModel(@NonNull final Navigator navigator) {
         this.navigator = get(navigator);
     }
 
     void openSearch() {
-        openSearch.onNext(Unit.DEFAULT);
+        openSearchSubject.onNext(Unit.DEFAULT);
     }
 
     @Override
     public void bind(@NonNull final CompositeDisposable disposables) {
-        disposables.add(openSearch.subscribeOn(mainThread())
-                                  .observeOn(computation())
-                                  .subscribe(__ -> navigator.openSearch(),
-                                             e -> Timber.e(e, "Error clearing search")));
+        disposables.add(openSearchSubject.observeOn(mainThread())
+                                         .subscribe(__ -> navigator.openSearch(),
```

Comment:
If we are going to change the thread to main here, Should we add an `AndroidPreconditions.mainThread()` style check to the Navigator methods? Evidently it didn't matter to the `Navigator` which thread it was invoked from before, although possibly there was a race condition if attempting to navigate when the Activity was being destroyed.